src/server/api/controllers/batch-status.controller.js

```diff
@@ -0,0 +1,152 @@
+const knex = require('../../config/db');
+const Error = require('../lib/utils/http-error');
+const moment = require('moment');
+
+const getBatchesStages = async (batchId) => {
+  try {
+    const batchesStages = await knex('batches')
+      .join('crop_stages', 'crop_stages.fk_crop_id', '=', 'batches.fk_crop_id')
+      .select(
+        'crop_stages.name',
+        'crop_stages.duration',
+        'batches.seeding_date',
+      )
+      .where('batches.id', batchId);
+    if (batchesStages.length === 0) {
+      throw new Error(`incorrect entry with the id of ${batchId}`, 404);
+    }
+    return batchesStages;
+  } catch (error) {
+    return error.message;
+  }
+};
+
+const currentStage = async (batchId, batchesStages) => {
+  /* eslint-disable no-param-reassign */
+  batchesStages.reduce((sum, batch) => {
+    batch.duration += sum;
+    return batch.duration;
+  }, 0);
+  /* eslint-enable no-param-reassign */
+  const today = moment();
+  const seedingDate = moment(batchesStages[0].seeding_date);
+  const currentDay = today.diff(seedingDate, 'days') + 1;
+  const filterDurationLessThanToday = batchesStages.filter(
+    (batch) => batch.duration >= currentDay,
+  );
+  const currentStageName =
+    filterDurationLessThanToday.length > 0
+      ? filterDurationLessThanToday[0].name
+      : 'delivered';
+  if (currentStage.length === 0) {
+    throw new Error(`incorrect entry with the id of ${batchId}`, 404);
+  }
+  if (currentStageName === 'delivered') return { status: currentStageName };
+  return { status: currentStageName, day: currentDay };
+};
+
+const getDaysTilHarvest = async (batchesStages) => {
+  try {
+    const [harvestId] = await knex('crop_stages')
+      .select('id')
+      .where('name', 'harvest');
+    const durationDaysBeforeHarvest = batchesStages
+      .filter((stage) => stage.id < harvestId.id)
+      .reduce((durationSum, cropStage) => {
+        return durationSum + cropStage.duration;
+      }, 0);
+    const todayDate = moment();
+    const seedingDate = moment(batchesStages[0].seeding_date);
+    const harvestStartDate = seedingDate.add(durationDaysBeforeHarvest, 'days');
+    if (
+      moment(todayDate).format('DD-MM-YYYY') >=
+      moment(harvestStartDate).format('DD-MM-YYYY')
+    )
+      return `Harvest started`;
```

Comment:
We should avoid returning this kind of data format. If our function returns integers when data is available, we should probably return `null` when there are no days and in the frontend we know if the result is `null`, then the `Harvest started`